### PR TITLE
Added error handling for socket:connect,recv and send calls

### PR DESCRIPTION
**Pull request by deepikabhavnani**
Checking for errors in return type for connect / recv and send calls. In case of connect failure recv / send call should not be attempted.

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -77,22 +77,38 @@ void scan_demo(WiFiInterface *wifi)
...
     TCPSocket socket;
+    nsapi_error_t response;
 
     printf("Sending HTTP request to www.arm.com...\r\n");
 
     // Open a socket on the network interface, and create a TCP connection to www.arm.com
     socket.open(net);
-    socket.connect("www.arm.com", 80);
+    response = socket.connect("www.arm.com", 80);
+    if(0 != response) {
```
Comment (0xc0170) [inline main.cpp]: nit: `if(0 != response)` space after `if`

Comment (0xc0170) [inline main.cpp]: One question ,as we had this in our tests as well, should we try to connect multiple times or once is enough?

**Comment by 0xc0170** [comment]
@bulislaw Please review

File: main.cpp [hunk 2]
```diff
@@ -77,22 +77,38 @@ void scan_demo(WiFiInterface *wifi)
...
+        socket.close();
+        return;
+    }
 
     // Send a simple http request
     char sbuffer[] = "GET / HTTP/1.1\r\nHost: www.arm.com\r\n\r\n";
-    int scount = socket.send(sbuffer, sizeof sbuffer);
-    printf("sent %d [%.*s]\r\n", scount, strstr(sbuffer, "\r\n")-sbuffer, sbuffer);
+    response = socket.send(sbuffer, sizeof sbuffer);
+    if(response < 0) {
```
Comment (bulislaw) [inline main.cpp]: If we are fixing error checking we should also be making sure the whole buffer is transmitted, as socket send doesn't guarantee all the bytes are actually transmitted.

File: main.cpp (hunk 1 above)
Comment (deepikabhavnani) [inline main.cpp]: Since this is sample example, I don't see the need of retrials.

File: main.cpp (hunk 2 above)
Comment (deepikabhavnani) [inline main.cpp]: @bulislaw In case of partial message transmitted to server, shall we continue to finish the message or retry entire request?

Comment (bulislaw) [inline main.cpp]: Just continue with what wasn't send.

File: main.cpp (hunk 1 above)
Comment (bulislaw) [inline main.cpp]: I think whoever runs the example (eg CI) should retry it.

**Comment by deepikabhavnani** [comment]
@bulislaw - Please review

**Comment by bulislaw** [comment]
LGTM

**Pull request merged**